src/Features/CSharp/Portable/UseSimpleUsingStatement/UseSimpleUsingStatementDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,134 @@
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseSimpleUsingStatement
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class UseSimpleUsingStatementDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+        public UseSimpleUsingStatementDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.UseSimpleUsingStatementDiagnosticId,
+                   new LocalizableResourceString(nameof(FeaturesResources.Use_simple_using_statement), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
+                   new LocalizableResourceString(nameof(FeaturesResources.using_statement_can_be_simplified), FeaturesResources.ResourceManager, typeof(FeaturesResources)))
+        {
+        }
+
+        public override bool OpenFileOnly(Workspace workspace) => false;
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterSyntaxNodeAction(AnalyzeSyntax, SyntaxKind.UsingStatement);
+
+        private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var usingStatement = (UsingStatementSyntax)context.Node;
+
+            if (!(usingStatement.Parent is BlockSyntax parentBlock))
+            {
+                // Don't offer on a using statement that is parented by another using statement.
+                // We'll just offer on the topmost using statement.
+                return;
+            }
+
+            // Check that all the immediately nested usings are convertible as well.  
+            // We don't want take a sequence of nested-using and only convert some of them.
+            for (var current = usingStatement; current != null; current = current.Statement as UsingStatementSyntax)
+            {
+                if (current.Declaration == null)
+                {
+                    return;
+                }
+            }
+
+            // Verify that changing this using-statement into a using-declaration will not
+            // change semantics.
+            if (!PreservesSemantics(parentBlock, usingStatement))
+            {
+                return;
+            }
+
+            var syntaxTree = context.Node.SyntaxTree;
+            var options = (CSharpParseOptions)syntaxTree.Options;
+            if (options.LanguageVersion < LanguageVersion.CSharp8)
+            {
+                return;
+            }
+
+            var cancellationToken = context.CancellationToken;
+            var optionSet = context.Options.GetDocumentOptionSetAsync(syntaxTree, cancellationToken).GetAwaiter().GetResult();
+            if (optionSet == null)
+            {
+                return;
+            }
+
+            var option = optionSet.GetOption(CSharpCodeStyleOptions.PreferStaticLocalFunction);
```

Comment:
How does this relate to simple using statements?

Reply:
Great catch!  Also a test hole.  Will fix!

Reply:
Fixed